Tiltfile

```diff
@@ -293,13 +310,48 @@ def enable_provider(name, debug):
             links = links,
         )
 
-def find_manager(yaml):
-    manifests = decode_yaml_stream(yaml)
+        provider_yaml, object_name = get_provider(manifests)
+        k8s_yaml(blob(provider_yaml))
+        k8s_resource(
+            objects = [object_name],
+            resource_deps = ["provider_crd"],
+            new_name = label.lower() + "_cr",
+            labels = [label],
+        )
+
+def find_manager(manifests):
     for m in manifests:
         if m["kind"] == "Deployment":
             return m["metadata"]["name"]
     return ""
 
+# This code reimplements cmd/clusterctl/client/repository/components.InventoryObject
+# It takes all required information from the Namespace resource.
+def get_provider(manifests):
+    for m in manifests:
+        if m["kind"] == "Namespace":
+            target_namespace = m["metadata"]["name"]
+            manifest_label = m["metadata"]["labels"]["cluster.x-k8s.io/provider"]
+
+    # Calculate provider type and name from manifest_label.
+    # Examples for manifest labels: "cluster-api", "infrastructure-docker",
+    # "bootstrap-kubeadm", "control-plane-kubeadm"
+    # type is based on the prefix.
+    # name is the manifest label without the provider type prefix.
+    type = "CoreProvider"
+    name = manifest_label
+    if manifest_label.startswith("infrastructure"):
+        type = "InfrastructureProvider"
+        name = manifest_label[len("infrastructure-"):]
+    if manifest_label.startswith("bootstrap"):
+        type = "BootstrapProvider"
+        name = manifest_label[len("bootstrap-"):]
+    if manifest_label.startswith("control-plane"):
+        type = "ControlPlaneProvider"
+        name = manifest_label[len("control-plane-"):]
+
+    return providersTemplate.format(manifest_label, manifest_label, target_namespace, name, type), "{}:Provider:{}".format(manifest_label, target_namespace)
```

Comment:
If we are dropping k8s_resource above, we can simplify here and return only the template

Reply:
I need the resource to define the dependency to the CRD. Otherwise the deployment of the provider CR will fail.